### DISCUSSION TITLE
什么条件下会生成带有一串数字后缀的文件

### DISCUSSION THREAD

**Discussion by chaoyue1217** [opensumi/core#3264] [Q&A] [answered]
![image](https://github.com/opensumi/core/assets/18662948/0e09bb1b-8ebb-4830-8400-17acf69b9e96)
如图，会出现一些带有数字后缀的文件，但是复现步骤不确定，所以想请教一下是否知道这种文件的创建时机

**Comment by erha19** [comment, marked as answer]
@chaoyue1217 这里估计是文件服务的逻辑造成了，因为部分系统环境下，直接覆盖文件内容可能出现权限问题，故当前采用的文件写策略是通过，新建临时文件写入内容，覆盖旧文件方式，该逻辑会造成临时文件的生成，详细代码见：

https://github.com/opensumi/core/blob/d7a79bab960c3d7459930af83bb8ad023426e0b8/packages/file-service/src/node/disk-file-system.provider.ts#L253
